tests/unit/utils/isSlotPopulated.spec.js

```diff
@@ -0,0 +1,95 @@
+import { mount } from '@vue/test-utils'
+
+import isSlotPopulated from '../../../src/utils/isSlotPopulated.js'
+
+
+const IsSlotPopulatedTest = {
+	name: 'IsSlotPopulatedTest',
+	template: '<div><slot /></div>',
+	computed : {
+		isDefaultPopulated() {
+			return isSlotPopulated(this.$slots.default?.())
+		},
+	},
+}
+
+let wrapper
+describe('isSlotPopulated.js', () => {
+	'use strict'
+	describe('when default slot contains empty array', () => {
+		it('shows empty slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: [],
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(false)
+		})
+	})
+
+	describe('when default slot contains empty string', () => {
+		it('shows empty slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: '',
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(false)
+		})
+	})
+
+	describe('when default slot contains string', () => {
+		it('shows populated slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: 'Test',
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(true)
+		})
+	})
+
+	describe('when default slot contains string with whitespace only', () => {
+		it('shows empty slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: '           ',
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(false)
+		})
+	})
+
+	describe('when default slot contains array containing string', () => {
+		it('shows populated slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: ['Test'],
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(true)
+		})
+	})
+
+	describe('when default slot contains comment and text', () => {
+		it('shows populated slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: '<!-- Commment -->Test',
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(true)
+		})
+	})
+
+	describe('when default slot contains comment only', () => {
+		it('shows empty slot', () => {
+			wrapper = mount(IsSlotPopulatedTest, {
+				slots: {
+					default: '<!-- Commment -->',
+				},
+			})
+			expect(wrapper.vm.isDefaultPopulated).toBe(false)
+		})
+	})
+})
```

Comment:
To be honest, the purpose of wrapping each test in a `describe` block is also not clear to me...

```suggestion
describe('isSlotPopulated.js', () => {
	it('is not populated when a slot contains an empty array', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: [],
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(false)
	})
	it('is not populated when a slot with an empty string', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: '',
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(false)
	})
	it('is populated when a slot contains a string', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: 'Test',
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(true)
	})
	it('is not populated when a slot contains a string with whitespace only', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: '           ',
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(false)
	})
	it('is populated when a slot contains an array containing a string', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: ['Test'],
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(true)
	})
	it('is populated when a slot contains a comment and a text', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: '<!-- Commment -->Test',
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(true)
	})
	it('is not populated when a slot contains a comment only', () => {
		const wrapper = mount(IsSlotPopulatedTest, {
			slots: {
				default: '<!-- Commment -->',
			},
		})
		expect(wrapper.vm.isDefaultPopulated).toBe(false)
	})
})
```

Other minor changes:
- removed "default" because it works with any slot
- rewrite it, `isSlotPopulated` doesn't "show" a slot but check if it is populated

Reply:
Fine with me.